### PR TITLE
JR2-225 Fix connection factory to support https

### PR DESCRIPTION
**Pull request by michael-wirth**

### CODE REVIEW

File: src/main/java/ch/admin/seco/jobroom/config/DefaultApacheHttpClientConnectionManagerFactory.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                    sslContext, NoopHostnameVerifier.INSTANCE));
+            } else {
+                sslContext.init(null, null, new SecureRandom());
+                registryBuilder.register(HTTPS_SCHEME, new SSLConnectionSocketFactory(sslContext));
+            }
+        } catch (NoSuchAlgorithmException e) {
+            LOG.warn("Error creating SSLContext", e);
+        } catch (KeyManagementException e) {
+            LOG.warn("Error creating SSLContext", e);
+        }
```
Comment (varusakov) [inline src/main/java/ch/admin/seco/jobroom/config/DefaultApacheHttpClientConnectionManagerFactory.java]: You can use one one catch block`catch (NoSuchAlgorithmException | KeyManagementException e) {LOG.warn("Error creating SSLContext", e);}`

File: src/main/java/ch/admin/seco/jobroom/config/DefaultApacheHttpClientConnectionManagerFactory.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        int maxTotalConnections, int maxConnectionsPerRoute, long timeToLive,
+        TimeUnit timeUnit, RegistryBuilder registryBuilder) {
+        if (registryBuilder == null) {
+            registryBuilder = RegistryBuilder.<ConnectionSocketFactory>create()
+                .register(HTTP_SCHEME, PlainConnectionSocketFactory.INSTANCE);
+        }
+        try {
+            final SSLContext sslContext = SSLContext.getInstance("SSL");
+            if (disableSslValidation) {
+                sslContext.init(null, new TrustManager[] {new X509TrustManager() {
```
Comment (varusakov) [inline src/main/java/ch/admin/seco/jobroom/config/DefaultApacheHttpClientConnectionManagerFactory.java]: I propose to extract `X509TrustManager` from function body to inner class

Comment (michael-wirth) [reply src/main/java/ch/admin/seco/jobroom/config/DefaultApacheHttpClientConnectionManagerFactory.java]: I copied the original file from spring-cloud-commons project and just fix the bug without a refactoring

Comment (michael-wirth) [reply src/main/java/ch/admin/seco/jobroom/config/DefaultApacheHttpClientConnectionManagerFactory.java]: fixed

File: src/main/java/ch/admin/seco/jobroom/config/DefaultApacheHttpClientConnectionManagerFactory.java (hunk 1 above)
Comment (michael-wirth) [reply src/main/java/ch/admin/seco/jobroom/config/DefaultApacheHttpClientConnectionManagerFactory.java]: fixed

File: src/main/java/ch/admin/seco/jobroom/config/DefaultApacheHttpClientConnectionManagerFactory.java (hunk 2 above)
Comment (varusakov) [reply src/main/java/ch/admin/seco/jobroom/config/DefaultApacheHttpClientConnectionManagerFactory.java]: Wasn't aware that it was file from spring-cloud-commons but thanks for fixing!

**Pull request merged**